app/post_metadata.go

```diff
@@ -417,8 +417,8 @@ func getLinkMetadataFromCache(requestURL string, timestamp int64) (*opengraph.Op
 }
 
 func (a *App) getLinkMetadataFromDatabase(requestURL string, timestamp int64) (*opengraph.OpenGraph, *model.PostImage, bool) {
-	result := <-a.Srv.Store.LinkMetadata().Get(requestURL, timestamp)
-	if result.Err != nil {
+	result, err := a.Srv.Store.LinkMetadata().Get(requestURL, timestamp)
```

Comment:
This is wrong, we no longer expect a result, probably here we can use variable name `linkMetadata`, and some lines later, you should use something like `data := linkMetadata.Data`